sources/web/datalab/polymer/components/datalab-sessions/datalab-sessions.ts

```diff
@@ -103,25 +94,31 @@ class SessionsElement extends Polymer.Element {
    * updates the _sessionList property.
    */
   _fetchSessionList() {
-    const self = this;
-    self._fetching = true;
+    // Don't overlap fetch requests. This can happen because we set up fetch from several sources:
+    // - Initialization in the ready() event handler.
+    // - Refresh mechanism called by the setInterval().
+    // - User clicking refresh button.
+    // - Sessions page gaining focus.
+    if (this._fetching) {
+      return Promise.resolve(null);
```

Comment:
I am suggesting it may be better for this function not to return a Promise. That's why I asked the question: when does it make sense for a caller to wait on this promise? The function might return right away with a no-op promise, so the caller can't rely on anything having actually happened before continuing, so why ever wait to continue?
Alternatively, set up the promise on line 103 so that it does not resolve until this._fetching gets cleared.